pkg/resources/scim_integration.go

```diff
@@ -74,140 +86,170 @@ func SCIMIntegration() *schema.Resource {
 	client := meta.(*provider.Context).Client
-	db := client.GetConn().DB
 	name := d.Get("name").(string)
+	id := sdk.NewAccountObjectIdentifier(name)
+	enabled := d.Get("enabled").(bool)
+	scimClient := d.Get("scim_client").(string)
+	runAsRole := d.Get("run_as_role").(string)
 
-	stmt := snowflake.NewSCIMIntegrationBuilder(name).Create()
-
-	// Set required fields
-	stmt.SetRaw(`TYPE=SCIM`)
-	stmt.SetString(`SCIM_CLIENT`, d.Get("scim_client").(string))
-	stmt.SetString(`RUN_AS_ROLE`, d.Get("provisioner_role").(string))
+	req := sdk.NewCreateScimSecurityIntegrationRequest(id, enabled, sdk.ScimSecurityIntegrationScimClientOption(scimClient), sdk.ScimSecurityIntegrationRunAsRoleOption(runAsRole))
 
-	// Set optional fields
-	if _, ok := d.GetOk("network_policy"); ok {
-		stmt.SetString(`NETWORK_POLICY`, d.Get("network_policy").(string))
+	// Set optionals
+	if v, ok := d.GetOk("network_policy"); ok {
+		req.WithNetworkPolicy(sdk.NewAccountObjectIdentifier(v.(string)))
 	}
-
-	if err := snowflake.Exec(db, stmt.Statement()); err != nil {
-		return fmt.Errorf("error creating security integration")
+	req.WithSyncPassword(d.Get("sync_password").(bool))
+	if v, ok := d.GetOk("comment"); ok {
+		req.WithComment(v.(string))
+	}
+	if err := client.SecurityIntegrations.CreateScim(ctx, req); err != nil {
+		return diag.FromErr(err)
 	}
 
-	d.SetId(name)
+	d.SetId(helpers.EncodeSnowflakeID(id))
 
-	return ReadSCIMIntegration(d, meta)
+	return ReadContextSCIMIntegration(ctx, d, meta)
 }
 
-// ReadSCIMIntegration implements schema.ReadFunc.
-func ReadSCIMIntegration(d *schema.ResourceData, meta interface{}) error {
+// ReadContextSCIMIntegration implements schema.ReadFunc.
+func ReadContextSCIMIntegration(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*provider.Context).Client
-	db := client.GetConn().DB
-	id := d.Id()
-
-	stmt := snowflake.NewSCIMIntegrationBuilder(id).Show()
-	row := snowflake.QueryRow(db, stmt)
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
 
-	// Some properties can come from the SHOW INTEGRATION call
-
-	s, err := snowflake.ScanScimIntegration(row)
+	integration, err := client.SecurityIntegrations.ShowByID(ctx, id)
 	if err != nil {
-		return fmt.Errorf("could not show security integration")
+		log.Printf("[DEBUG] notification integration (%s) not found", d.Id())
+		d.SetId("")
+		return diag.FromErr(err)
 	}
 
-	// Note: category must be Security or something is broken
-	if c := s.Category.String; c != "SECURITY" {
-		return fmt.Errorf("expected %v to be an Security integration, got %v", id, c)
+	if c := integration.Category; c != "SECURITY" {
+		return diag.FromErr(fmt.Errorf("expected %v to be a SECURITY integration, got %v", id, c))
 	}
 
-	if err := d.Set("scim_client", strings.TrimPrefix(s.IntegrationType.String, "SCIM - ")); err != nil {
-		return err
+	if err := d.Set("name", integration.Name); err != nil {
```

Comment:
Recently I was wondering if we want to set the `name` field. To read this integration we searched by `name`, so It should be the same and the output from Snowflake may differ in casing/quoting which may result in plan being produced for `name`.

Reply:
Yesterday we talked about `name`s (again ;_;) and shouldn't we actually have them in in Read, so they'll be read during import?

Reply:
Yeah, imports aren't working correctly because of that. Reverted this and removed skip check.